app/main_2.py

```diff
@@ -0,0 +1,79 @@
+class Dictionary:
+    def __init__(self, size=8):
+        self.storage = [[] for _ in range(size)]
+        self.size = size
+        self.length = 0
+        self.load_factor = 2 / 3
+
+    def __setitem__(self, key, value):
+        storage_idx = hash(key) % self.size
+        for element in self.storage[storage_idx]:
```

Comment:
This loop takes O(n) time for every search of the missing key. You should optimize it :(